src/query/generated/proto/rpcpb/query.proto

```diff
@@ -90,3 +92,44 @@ message M3Segment {
 	int64 startTime = 3;
 	int64 blockSize = 4;
 }
+
+message SearchRequest {
+	oneof matchers {
+		TagMatchers tagMatchers     = 1;
+	}
+	SearchType type               = 2;
+	repeated bytes filterNameTags = 3;
+}
+
+enum SearchType {
+	// Requests tag names and values
+	DEFAULT = 0;
+	// Requests tag names only
+	TAGNAME = 1;
+}
+
+message SearchResponse {
+	oneof value {
+		M3TagProperties compressed = 1;
+		TagProperties decompressed = 2;
+	}
+}
+
+message M3TagProperties {
+	repeated M3TagProperty properties = 1;
+}
+
+message M3TagProperty {
+	bytes id             = 1;
+	bytes compressedTags = 2;
+}
+
+message TagProperties {
+	repeated TagProperty properties = 1;
+}
+
+message TagProperty {
+	bytes key             = 1;
+	uint32 weight         = 2;
```

Comment:
I don't think we need weight for now (we were only getting that because it was provided), can we remove this and add it back when we have a use for it?

Reply:
Nice :)

Actually had it deleted in latest WIP so works out